File: src/my_project/easy_problems/from151To200/check_all_characters_have_equal_number.py
from typing import List 


class Solution:
    def areOccurrencesEqual(self, s: str) -> bool:

        temp_dict = dict()
        flag: str = True

        for letter in s: 
            temp_dict[letter] = 0 

        for letter in s:
            temp_dict[letter] += 1

        for i in range(1,len(s)):
            if temp_dict[s[i-1]] !=  temp_dict[s[i]]:
                flag = False 
                return flag


        return flag 

solution = Solution()

print(solution.areOccurrencesEqual(s = "abacbc"))

print(solution.areOccurrencesEqual(s = "aaabb"))

File: src/my_project_typescript/easy_problems_typescript/from51To100/checkAllCharactersHaveEqualNumber.ts
function areOccurrencesEqual(s: string): boolean {

    let tempDict: any = {}
    let flag: boolean = true 

    for (let letter of s){
        tempDict[letter] = 0
    }

    for (let letter of s){
        tempDict[letter] += 1
    }

    for (let i = 1; i < s.length; i++){
        if (tempDict[s[i-1]] !== tempDict[s[i]]){
            flag = false 
            return flag 
        }
    }
    return flag 
};


console.log(areOccurrencesEqual("abacbc"))

console.log(areOccurrencesEqual("aaabb"))
